### PR TITLE
Sway 1.0

### PR DESCRIPTION
**Pull request by andypost**
- https://github.com/swaywm/sway/releases/tag/1.0
- https://github.com/swaywm/wlroots/releases

### CODE REVIEW

**Comment by andypost** [comment]
> >>> WARNING: wlroots-dev*: Could not find any provider for pc:wayland-protocols>=1.17

**Comment by TBK** [comment]
Strange, the file is there - https://pkgs.alpinelinux.org/contents?branch=edge&name=wayland-protocols&arch=x86&repo=main

@ddevault Congratulations on the 1.0 release 👍

**Comment by ddevault** [comment]
Thanks ^^

Was going to send this patch along myself shortly, cheers.

**Comment by TBK** [comment]
Maybe it is time to move it from testing to community?

@ddevault do you want to claim maintainership?

**Comment by ddevault** [comment]
Henrik has been doing a fine job, but if he'd like me to take over that's fine too.

+1 to moving to community

**Comment by TBK** [comment]
Ahh I just assumed since `# Maintainer:` is empty. @HRio what do you say to the move?

**Comment by HRio** [comment]
@ddevault do you feel wlroots is "ready" for a move to community/? i.e. 0.5.0 had breaking changes.

At the moment we only have swaywm using it, but if we end up having more window managers depending on wlroots, breaking changes to wlroots (and considering its not in testing/) will take some coordination to handle, without collateral damage...

**Comment by ddevault** [comment]
Future wlroots releases will have breaking changes, too. We update the soname accordingly, though.

**Comment by HRio** [comment]
3.10 is few months away, maybe we should leave it in testing/ for now and then reconsider the move to community/ as we get closer to 3.10? that will give wlroots some more time to stabilize and we also know better if we have more wlroots based window managers to consider.

**Comment by ddevault** [comment]
wlroots is going to take several years to stabilize.

**Comment by TBK** [comment]
I have just send a mail to abi-laboratory.pro to get wlroots added.

Would be great if we could get an Sway equivalent site of https://wiki.alpinelinux.org/wiki/XFCE_Setup

**Comment by ddevault** [comment]
https://wiki.alpinelinux.org/wiki/Sway

**Comment by TBK** [comment]
I just noticed :S It is just not as extensive.

**Comment by ddevault** [comment]
*shrug*

I'm satisfied with it.

**Comment by HRio** [comment]
@ddevault I would be very happy if you would like to maintain wlroots and sway in Alpine.

(I have not added myself as maintainer as I have a have a bit limited time to maintain more stuff)

**Comment by ddevault** [comment]
Sure, @andypost can you add me to the APKBUILD?

**Comment by HRio** [comment]
the build looks as bit strange
```log
Dependency wlroots found: YES 0.5.0
994	Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
995	Dependency scdoc found: YES 1.9.4
996	Program /usr/local/bin/scdoc found: NO
997	
998	meson.build:100:1: ERROR: Program(s) ['/usr/local/bin/scdoc'] not found or not executable
999	
1000	A full log can be found at /home/buildozer/drone/aports/testing/sway/src/sway-1.0/output/meson-logs/meson-log.txt
1001	>>> ERROR: sway: build failed
```

**Comment by HRio** [comment]
a sway 1.0-rc5 build to compare
```log
Dependency wlroots found: YES 0.4.1
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Dependency scdoc found: YES 1.9.4
Program scdoc found: YES (/usr/bin/scdoc)
Program sh found: YES (/bin/sh)
Configuring config.h using configuration
```

**Comment by ddevault** [comment]
Looks like a bug in the scdoc package. A recent scdoc version introduced a pkg-config file with the path to scdoc, and the prefix is wrong.

**Comment by HRio** [comment]
yes
```
cat /usr/lib/pkgconfig/scdoc.pc                                                                                                                           [~/git/aports/testing/sway] [REF-master]
prefix=/usr/local
exec_prefix=${prefix}
scdoc=${exec_prefix}/bin/scdoc

Name: scdoc
Description: Man page generator
Version: 1.9.4
```

**Comment by HRio** [comment]
fixed in scdoc 1.9.4-r1

**Comment by andypost** [comment]
Please give a diff to add maintainers

File: testing/sway/APKBUILD [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Contributor: Henrik Riomar <henrik.riomar@gmail.com>
 # Maintainer:
 pkgname=sway
-pkgver=1.0_rc5
-_pkgver=1.0-rc5
+pkgver=1.0
+_pkgver=1.0
 pkgrel=1
```
Comment (HRio) [inline testing/sway/APKBUILD]: reset to 0

**Comment by HRio** [comment]
> Please give a diff to add maintainers

@ddevault I can commit this PR first and the you can send a patch via patchwork, that I will commit.

**Comment by ddevault** [comment]
+1, just let me know when

File: testing/sway/APKBUILD (hunk 1 above)
Comment (andypost) [reply testing/sway/APKBUILD]: Fixed pkgrel

**Comment by HRio** [comment]
> +1, just let me know when

@ddevault now its committed: https://git.alpinelinux.org/aports/log/

**Comment by ddevault** [comment]
Patches sent

**Pull request closed without merging**